packages/dds/tree/src/test/feature-libraries/chunked-forest/uniformChunk.spec.ts

```diff
@@ -0,0 +1,266 @@
+        );
+        for (const [k, v] of info.shape.fields) {
+            for (let index = 0; index < v.topLevelLength; index++) {
+                // TODO: if we keep all the duplicated position info, inline positionIndex into field offsets to save the addition.
+                const offset = v.offset + index * v.shape.positions.length;
+                const element = shape.positions[offset + positionIndex];
+                assert.equal(element.parentIndex, index);
+                assert.equal(element.parentField, k);
+                assert.equal(element.parent, info);
+            }
+        }
+    });
+}
+
+// testing is per node, and our data can have multiple nodes at the root, so split tests as needed:
+const testData: TestTree<[number, TreeChunk]>[] = testTrees.flatMap(({ name, data, reference }) => {
+    const out: TestTree<[number, TreeChunk]>[] = [];
+    for (let index = 0; index < reference.length; index++) {
+        out.push({
+            name: reference.length > 1 ? `${name} part ${index + 1}` : name,
+            data: [index, data],
+            reference: reference[index],
+            path: { parent: undefined, parentIndex: index, parentField: dummyRoot },
+        });
+    }
+    return out;
+});
+
+describe("uniformChunk", () => {
+    describe("shapes", () => {
+        for (const tree of testTrees) {
+            it(`validate shape for ${tree.name}`, () => {
+                validateShape((tree.data as UniformChunk).shape);
+            });
+        }
+    });
+
+    testSpecializedCursor<[number, TreeChunk], ITreeCursorSynchronous>({
+        cursorName: "uniformChunkCursor",
+        builders: {
+            withKeys: (keys) => {
+                const schema: TreeSchemaIdentifier = brand("fakeSchema");
+                const withKeysShape = new TreeShape(
+                    schema,
+                    false,
+                    keys.map((key) => [key, emptyShape, 1] as const),
+                );
+                return [0, uniformChunk(withKeysShape.withTopLevelLength(1), [])];
+            },
+        },
+        cursorFactory: (data: [number, TreeChunk]): ITreeCursorSynchronous => {
+            const cursor = data[1].cursor();
+            assert(cursor.seekNodes(data[0]));
+            return cursor;
+        },
+        testData,
+    });
+
+    const cursorSources = [
+        { name: "uniformChunk", factory: (data: TreeChunk) => data.cursor() },
+        {
+            name: "jsonable",
+            factory: (data: TreeChunk) => singleTextCursor(jsonableTreeFromCursor(data.cursor())),
+        },
+        {
+            name: "mapTree",
+            factory: (data: TreeChunk) => singleMapTreeCursor(mapTreeFromCursor(data.cursor())),
+        },
+        {
+            name: "json",
+            factory: (data: TreeChunk) => singleJsonCursor(cursorToJsonObject(data.cursor())),
+        },
+    ];
+
+    for (const { name: cursorName, factory } of cursorSources) {
```

Comment:
This seems redundant with the `jsonCursor.bench.ts` file. Can we merge the two?

Reply:
We have different limitations on the shapes of the data that we can handler here, so most of our existing benchmarks can't be run with UniformChunk Cursors. While it may be possible to partially combine them, I don't see a clean way to share much more that is already shared (ex: the sum function is already shared) without losing flexibility. These benchmarks carefully pick test data that happens to work with JsonCursor and UniformChunks, neither of which are very general, and if we can't share the test data cases, or the selection of which cursors to compare to, I don't see a lot more worth sharing.

Once we have a cursor that combines multiple chunk formats and can handle arbitrary data, that will work great with the other test suite and can be added to it.